releases/release-5.1.4.md

```diff
@@ -0,0 +1,165 @@
+---
+title: TiDB 5.1.4 Release Notes
+category: Releases
+---
+
+# TiDB 5.1.4 Release Notes
+
+Release Date: xx, 2022
+
+TiDB version: 5.1.4
+
+## Compatibility changes
+
```

Comment:
- Change the default value of system variable [`tidb_analyze_version`](https://docs.pingcap.com/tidb/v5.1/system-variables#tidb_analyze_version-new-in-v510) to 1. [#31748](https://github.com/pingcap/tidb/issues/31748)